### PR TITLE
[LWG  2] P2789R0 Ready and Tentatively Ready issues

### LINKED ISSUE #6092: [2023-02 LWG Motion 2] P2789R0 C++ Standard Library Issues to be moved in Issaquah, Feb. 2023

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #6092.
Fixes cplusplus/papers#1478
Fixes cplusplus/nbballot#534

### CODE REVIEW

File: source/ranges.tex [hunk 1]
```diff
@@ -2298,7 +2300,10 @@
 \item
 \tcode{\libconcept{common_range}<R>} is \tcode{true},
 \item
-\tcode{\exposconcept{cpp17-input-iterator}<iterator_t<R>>} is \tcode{true}, and
+if the \grammarterm{qualified-id}
```
Comment (JohelEGP) [inline source/ranges.tex:2303]:
```suggestion
the \grammarterm{qualified-id}
```
Not in the paper. Reads awkward like #6113. Unlike #6113, this item has no condition; rather, this item is a term in a conjunction, so removing 'if' suffices.

File: source/memory.tex [hunk 2]
```diff
@@ -135,6 +135,9 @@
   template<class T, class Alloc, class U, class V>
     constexpr auto uses_allocator_construction_args(const Alloc& alloc,             // freestanding
                                                     const pair<U, V>& pr) noexcept;
+  template<class T, class Alloc, @\exposconcept{pair-like}@ P>
+    constexpr auto uses_allocator_construction_args(const Alloc& alloc,             // freestanding
+                                                    P&& p) noexcept;
   template<class T, class Alloc, class U, class V>
     constexpr auto uses_allocator_construction_args(const Alloc& alloc,             // freestanding
                                                     pair<U, V>&& pr) noexcept;
```
Comment (JohelEGP) [inline source/memory.tex:143]: The added signature is requested to be inserted before the signature
```C++
  template<class T, class Alloc, class U>
    constexpr auto uses_allocator_construction_args(const Alloc& alloc, U&& u) noexcept;
```

File: source/ranges.tex [hunk 3]
```diff
@@ -10565,6 +10579,10 @@
     adjacent_view() requires @\libconcept{default_initializable}@<V> = default;
     constexpr explicit adjacent_view(V base);
 
+
```
Comment (JohelEGP) [inline source/ranges.tex:10582]:
```suggestion
```

File: source/ranges.tex [hunk 4]
```diff
@@ -12119,6 +12147,29 @@
...
+\begin{itemdecl}
+friend constexpr void iter_swap(const @\exposid{inner-iterator}@& x, const @\exposid{inner-iterator}@& y)
+  noexcept(noexcept(ranges::iter_swap(*x.@\exposid{parent_}@->@\exposid{current_}@, *y.@\exposid{parent_}@->@\exposid{current_}@)))
+  requires @\libconcept{indirectly_swappable}@<iterator_t<V>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{ranges::iter_swap(*x.\exposid{parent_}->\exposid{current_}, *y.\exposid{parent_}->\exposid{current_});}
```
Comment (JohelEGP) [inline source/ranges.tex:12170]:
```suggestion
Equivalent to: \tcode{ranges::iter_swap(*x.\exposid{parent_}->\exposid{current_}, *y.\exposid{parent_}->\exposid{current_})}.
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/locales.tex [hunk 5]
```diff
@@ -587,7 +591,7 @@
 \begin{itemdescr}
 \pnum
 \effects
-The same as \tcode{locale(other, std_name.c_str(), cat)}.
+Equivalent to \tcode{locale(other, std_name.c_str(), cat)}.
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/locales.tex:594]: Thanks!

File: source/ranges.tex (hunk 4 above)
Comment (burblebee) [reply source/ranges.tex:12170]: Actually no; when we introduce code with "Equivalent to:" (note the trailing ":"), the code is supposed to be complete, as if in a codeblock that could be substituted. Your suggestion would apply if the code had been introduced using "Equivalent to", the idea being that later reads as a statement. That said, if this should be an expression and not a statement, then the ";" should be removed. Alternatively, we could use your suggestion and introduce it with "Equivalent to".  What do you prefer?

File: source/ranges.tex (hunk 1 above)
Comment (burblebee) [reply source/ranges.tex:2303]: It didn't read correctly to me either, but wasn't sure what was actually intended here (like, is there additional text missing?).  Added as a separate commit.

File: source/ranges.tex (hunk 4 above)
Comment (JohelEGP) [reply source/ranges.tex:12169]: Ah, yes. Again, I forgot to remove the colon, too. Edited.

**Review by JohelEGP — APPROVED** [review]

File: source/ranges.tex (hunk 4 above)
Comment (tkoeppe) [reply source/ranges.tex:12169]: I'm fine leaving it as proposed. We could make the change, but we don't _have_ to, and that way we retain some symmetry between the two instances.

**Pull request merged**